src/bluesearch/entrypoint/database/run.py

```diff
@@ -0,0 +1,636 @@
+        "All papers from the given month until today will be downloaded.",
+    )
+    parser.add_argument(
+        "--filter-config",
+        required=True,
+        type=Path,
+        help="""
+        Path to a .JSONL file that defines all the rules for filtering.
+        """,
+    )
+    parser.add_argument(
+        "--output-dir",
+        required=True,
+        type=Path,
+        help="""
+        Path to the output folder. All the results stored under
+        `output_dir/source/date` where date is concatenation of the
+        `from_month` and the day of execution of this command.
+        """,
+    )
+    parser.add_argument(
+        "--db-url",
+        required=True,
+        type=str,
+        help="""
+        The location of the database depending on the database type.
+
+        For MySQL and MariaDB the server URL should be provided, for SQLite the
+        location of the database file. Generally, the scheme part of
+        the URL should be omitted, e.g. for MySQL the URL should be
+        of the form 'my_sql_server.ch:1234/my_database' and for SQLite
+        of the form '/path/to/the/local/database.db'.
+        """,
+    )
+    parser.add_argument(
+        "--db-type",
+        default="sqlite",
+        type=str,
+        choices=("mariadb", "mysql", "postgres", "sqlite"),
+        help="Type of the database.",
+    )
+    parser.add_argument(
+        "--mesh-topic-db",
+        type=Path,
+        help="""
+        The JSON file with MeSH topic hierarchy information. Mandatory for
+        source types "pmc" and "pubmed".
+
+        The JSON file should contain a flat dictionary with MeSH topic tree
+        numbers mapped to the corresponding topic labels. This file can be
+        produced using the `bbs_database parse-mesh-rdf` command. See that
+        command's description for more details.
+        """,
+    )
+    parser.add_argument(
+        "--dry-run",
+        "-n",
+        action="store_true",
+        help="Prints out a diagram of the pipeline without running it.",
+    )
+    parser.add_argument(
+        "--grobid-host",
+        type=str,
+        help="The host of the GROBID server.",
+    )
+    parser.add_argument(
+        "--grobid-port",
+        type=int,
+        help="The port of the GROBID server.",
+    )
+    parser.add_argument(
+        "--identifier",
+        type=str,
+        help="""Custom name of the identifier. If not specified, we use
+        `from-month_today`
+        """,
+    )
+    parser.add_argument(
+        "--final-task",
+        type=str,
+        help="""Name of the task where to manually stop the pipeline. Note
+        that the task itself will be included.
+        """,
+    )
+
+    return parser
+
+
+BBS_BINARY = ["bbs_database"]
+VERBOSITY = ["-v"]  # for the entrypoint subprocesses
+CAPTURE_OUTPUT = False
+IDENTIFIER = None  # make sure the same for all tasks
+
+
+class DownloadTask(ExternalProgramTask):
```

Comment:
We discussed this extensively with @EmilieDel and we actually went "full circle" when working on this  PR: `ExternalTask` -> `Task` -> `ExternalTask`. 

Let me point out some of the motivation behind using `ExternalTask`
* The `run` functions assume the input parameters have been parsed by the corresponding parser. So in order to use these `run` functions directly one needs to make sure the types are correct (=extra work). By using `ExternalTask` the parsing is done automatically.
* The startup time of `bbs_database command` is fast and there does not seem to be any major overhead of creating a subprocess
* In theory, the pipeline steps could be run in a different virtual env/docker container/image than this `luigi` pipeline. Of course this would require some extra work but just the logical separation that the pipeline runner process is different from the worker process felt like the right choice.
* This PR is a just a sketch.

Reply:
## Discussed on 2022-03-08

Looks fine with `ExternalTask` for the time being.

---- 
> The `run` functions assume the input parameters have been parsed by the corresponding parser. So in order to use these `run` functions directly one needs to make sure the types are correct (=extra work). By using `ExternalTask` the parsing is done automatically.

As a note, this could also be achieved by calling, e.g., 
```python
bluesearch.entrypoint.database.parent.main("download", "arg1", ...)
```
instead of 
```python
bluesearch.entrypoint.database.download.run(arg1, ...)
```